cirrus/server/tests/test_telemetry.py

```diff
@@ -0,0 +1,116 @@
+    assert len(snapshot) == 2
+
+    extra_1 = snapshot[0].extra
+    assert extra_1["app_id"] == "test_app_id"
+    assert extra_1["user_id"] == "test_client_id"
+    assert extra_1["experiment"] == "cirrus-test-1"
+    assert extra_1["branch"] == ""
+    assert extra_1["enrollment_id"] == "enrollment_id"
+    assert extra_1["experiment_type"] == RecipeType.ROLLOUT.value
+
+    extra_2 = snapshot[1].extra
+    assert extra_2["app_id"] == "test_app_id"
+    assert extra_2["user_id"] == "test_client_id"
+    assert extra_2["experiment"] == "cirrus-test-2"
+    assert extra_2["branch"] == "treatment-a"
+    assert extra_2["enrollment_id"] == "enrollment_id"
+    assert extra_2["experiment_type"] == RecipeType.EXPERIMENT.value
+
+
+def test_enrollment_metrics_recorded_with_record_metrics(recipes):
+    app.state.remote_setting.update_recipes(recipes)
+    _, metrics = initialize_glean()
+    enrolled_partial_configuration = {
+        # We don't need the other keys for this particular test
+        "events": [
+            {
+                "branch_slug": "",
+                "change": "Enrollment",
+                "enrollment_id": "enrollment_id",
+                "experiment_slug": "cirrus-test-1",
+                "reason": None,
+            },
+            {
+                "branch_slug": "treatment-a",
+                "change": "Enrollment",
+                "enrollment_id": "enrollment_id",
+                "experiment_slug": "cirrus-test-2",
+                "reason": None,
+            },
+            {
+                "branch_slug": "",
+                "change": "EnrollFailed",
+                "enrollment_id": "blah",
+                "experiment_slug": "fake-experiment",
+                "reason": "not_selected",
+            },
+        ],
+    }
+
+    app.state.pings.enrollment.test_before_next_submit(before_enrollment_ping)
+
+    record_metrics(enrolled_partial_configuration, "test_client_id")
+
+    assert app.state.metrics.cirrus_events.enrollment.test_get_value() is None
+
+
+@pytest.mark.asyncio
+async def test_enrollment_metrics_recorded_with_compute_features(recipes):
+    context = json.dumps(
+        {
+            "app_id": "org.mozilla.test",
+            "app_name": "test_app",
+            "channel": "release",
+        }
+    )
+    sdk = SDK(context=context)
+
+    request = FeatureRequest(
+        client_id="test-client-id", context={"user_id": "test-client-id"}
+    )
+
+    app.state.remote_setting.update_recipes(recipes)
+    sdk.set_experiments(json.dumps(recipes))
+    _, metrics = initialize_glean()
+
+    app.state.pings.enrollment.test_before_next_submit(before_enrollment_ping)
+
+    patch.object(
+        app.state.sdk,
+        "compute_enrollments",
+        return_value=sdk.compute_enrollments(
+            {"clientId": request.client_id, "requestContext": request.context}
+        ),
+    )
+
+    await compute_features(request)
```

Comment:
This is where the problem lies.

Reply:
do you need `result = await compute_features(request)`?

Reply:
Yep the `await` makes sure it actually finishes execution before moving forward